bokeh/util/dependencies.py

```diff
@@ -53,14 +54,20 @@ def import_required(mod_name, error_msg):
 
 def detect_phantomjs():
     '''Detect if PhantomJS is avaiable in PATH.'''
-    if settings.phantomjs_path() is not None:
-        phantomjs_path = settings.phantomjs_path()
-    else:
-        phantomjs_path = "phantomjs"
+
+    try:
+        phantomjs_path = shutil.which('phantomjs')
+    # Python 2 relies on Environment variable in PATH
+    except AttributeError:
+        if settings.phantomjs_path() is not None:
+            phantomjs_path = settings.phantomjs_path()
+        else:
+            phantomjs_path = "phantomjs"
```

Comment:
I think the env var setting should always be available as an explicit override, even in Python 3. Otherwise if it is `None`, then use `shutil.which`

Reply:
I'm amicable to that. I would anticipate it looking as follows:
```python 
 def detect_phantomjs():
      '''Detect if PhantomJS is avaiable in PATH.'''
      if settings.phantomjs_path() is not None:
           phantomjs_path = settings.phantomjs_path()
      else:
          try:
               phantomjs_path = shutil.which('phantomjs')
           # Python 2 relies on Environment variable in PATH - attempt to use as follows
          except AttributeError:
               phantomjs_path = "phantomjs"
```

Does that seem good to you?

Reply:
Yes that looks perfect